### PR TITLE
Constructing/Deconstructing Suit Storage Units

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Allows you to build and tear down Suit Storage Units.

Two types are able to be built: Standard and Industrial (think the ones that Engineering gets)

## Why It's Good For The Game
You should be able to move/decon these and replace them if theyre destroyed.

## Testing
Broke and made a buncha SSUs and ensured they dropped contents and such on decon.

## Changelog
:cl:
tweak: SSUs can now be constructed/deconstructed.
/:cl:

### CODE REVIEW

File: code/game/machinery/constructable_frame.dm [hunk 1]
```diff
@@ -1078,3 +1078,27 @@ to destroy them and players will be able to make replacements.
...
+	build_path = /obj/machinery/suit_storage_unit
+	board_type = "machine"
+	origin_tech = "materials=2;engineering=2"
+	req_components = list(
+							/obj/item/stock_parts/matter_bin = 1,
+							/obj/item/stock_parts/micro_laser = 1,
+							/obj/item/stack/cable_coil = 3,
+							/obj/item/stack/sheet/rglass = 5)
+
+/obj/item/circuitboard/suit_storage_unit/industrial
```
Comment (Henri215) [inline code/game/machinery/constructable_frame.dm:1094]: Why make it inherit suit_storage_unit board?

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -48,11 +48,13 @@
 	/// If set, turned into typecache in Initialize, other wise, defaults to mob/living typecache
 	var/list/occupant_typecache
 	var/atom/movable/occupant = null
+	var/board_type = /obj/item/circuitboard/suit_storage_unit
```
Comment (Henri215) [inline code/game/machinery/suit_storage_unit.dm:51]: Why is this necessary?

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -275,6 +279,26 @@
...
+		component_parts += new /obj/item/stack/sheet/plasteel(null)
+		component_parts += new /obj/item/stack/sheet/plasteel(null)
+		component_parts += new /obj/item/stack/sheet/plasteel(null)
+	else
+		component_parts += new /obj/item/stack/sheet/rglass(null)
+		component_parts += new /obj/item/stack/sheet/rglass(null)
+		component_parts += new /obj/item/stack/sheet/rglass(null)
+		component_parts += new /obj/item/stack/sheet/rglass(null)
+		component_parts += new /obj/item/stack/sheet/rglass(null)
+	RefreshParts() // MMTODO: needs "upgrades" for the different levels of parts.
```
Comment (Henri215) [inline code/game/machinery/suit_storage_unit.dm:300]: What does "MMTODO" means? I am curious

Comment (matttheficus) [reply code/game/machinery/suit_storage_unit.dm:300]: If i write just "TODO" i have to search through all of em everywhere. Writing code at 2am and then wanting to sleep hit me.

Comment (matttheficus) [reply code/game/machinery/suit_storage_unit.dm:300]: Wow the extenstion didnt work, anywho, see below.

File: code/game/machinery/constructable_frame.dm (hunk 1 above)
Comment (matttheficus) [reply code/game/machinery/constructable_frame.dm:1094]: The only difference between the two is the icon - one has a sliding metal door, the other is a glass.

File: code/game/machinery/suit_storage_unit.dm (hunk 3 above)
Comment (matttheficus) [reply code/game/machinery/suit_storage_unit.dm]: Update: not going to add any upgradeable bits for this, frankly. More an aesthetic/being able to move stuff around PR anyway.

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -337,6 +361,12 @@
 			to_chat(usr, "<span class='warning'>The unit is not operational.</span>")
 		return
 	if(panel_open)
+		if(istype(I, /obj/item/crowbar))
+			if(locked)
+				to_chat(user, "<span class='warning'>The security system prevents you from deconstructing [src]!</span>")
```
Comment (S34NW) [inline code/game/machinery/suit_storage_unit.dm:366]:
```suggestion
				to_chat(user, "<span class='warning'>The security system prevents you from deconstructing [src]!</span>")
				return
```
you didn't actually return here

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/game/machinery/suit_storage_unit.dm [hunk 5]
```diff
@@ -275,6 +279,26 @@
 
 /obj/machinery/suit_storage_unit/Initialize()
 	. = ..()
+
+	component_parts = list()
+	component_parts += new board_type(null)
+	component_parts += new /obj/item/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/stock_parts/micro_laser(null)
+	component_parts += new /obj/item/stack/cable_coil(null,3)
```
Comment (Henri215) [inline code/game/machinery/suit_storage_unit.dm:287]:
```suggestion
	component_parts += new /obj/item/stack/cable_coil(null, 3)
```

File: code/modules/research/designs/machine_designs.dm [hunk 6]
```diff
@@ -82,6 +82,26 @@
 	build_path = /obj/item/circuitboard/power_turbine
 	category = list ("Engineering Machinery")
 
+/datum/design/suit_storage_unit
+	name = "Machine Design (Suit Storage Unit)"
+	desc = "The circuit board for a Suit Storage Unit."
+	id = "ssu"
+	req_tech = list("engineering" = 3,"materials" = 3)
```
Comment (Henri215) [inline code/modules/research/designs/machine_designs.dm:89]:
```suggestion
	req_tech = list("engineering" = 3, "materials" = 3)
```

File: code/modules/research/designs/machine_designs.dm [hunk 7]
```diff
@@ -82,6 +82,26 @@
...
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/suit_storage_unit
+	category = list("Engineering Machinery")
+
+/datum/design/suit_storage_unit/industrial
+	name = "Machine Design (Industrial Suit Storage Unit)"
+	desc = "The circuit board for an Industrial Suit Storage Unit."
+	id = "issu"
+	req_tech = list("engineering" = 5,"materials" = 4)
```
Comment (Henri215) [inline code/modules/research/designs/machine_designs.dm:99]:
```suggestion
	req_tech = list("engineering" = 5, "materials" = 4)
```

**Review by S34NW — APPROVED** [review]
